### DISCUSSION TITLE
Keyboard shortcut to pin the active app

### DISCUSSION THREAD

**Discussion by maxwellhibbert** [rxhanson/RectanglePro-Community#575] [Ideas]
Hi, I've been loving pin mode, but I often change the app that's pinned. It'd be sweet to do that with a shortcut to pin the current active app rather than loading the menu.

**Comment by rxhanson** [comment]
Sorry that I never responded to this one, but thanks for the feedback! I'll add it to my backlog, but unfortunately can't promise a timeframe for when I can get to it.

**Comment by 5HT2** [comment]
You could honestly do something like this, but it would be a lot easier if we could just keybind it in Rectangle Pro.
https://forum.keyboardmaestro.com/t/applescript-access-menu-items-in-top-bar/23509

I would also really appreciate the ability to do so :)

---

Honestly you could probably forego using an applescript for that anyways and just trigger something like:
```bash
BUNDLE_ID="you should supply some way to get the active bundle ID here, either via applescript or otherwise"
defaults write "com.knollsoft.Hookshot" todoApplication "$BUNDLE_ID"
open -g "rectangle-pro://execute-action?name=pin"
open -g "rectangle-pro://execute-action?name=reflow-pin"
```

**Reply by rxhanson** [reply]
Note that Rectangle Pro would have to be restarted after the `defaults` command.

Thanks for bringing this one back up, good to know that there's more demand for it.

**Comment by 5HT2** [comment]
Somewhat related, it looks like pinning shown in the right click menu? Should I file a separate issue for that?

Edit: see #574
